src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -29,20 +29,20 @@
 /**
  * @author Samuel Roze <samuel.roze@gmail.com>
  */
-class Serializer implements SerializerInterface
+class Serializer implements FormatAndContextAwareSerializerInterface
 {
+    use FormatAndContextAwareSerializerTrait;
+
     public const MESSENGER_SERIALIZATION_CONTEXT = 'messenger_serialization';
     private const STAMP_HEADER_PREFIX = 'X-Message-Stamp-';
 
     private SymfonySerializerInterface $serializer;
-    private string $format;
-    private array $context;
 
     public function __construct(SymfonySerializerInterface $serializer = null, string $format = 'json', array $context = [])
     {
         $this->serializer = $serializer ?? self::create()->serializer;
-        $this->format = $format;
-        $this->context = $context + [self::MESSENGER_SERIALIZATION_CONTEXT => true];
+        $this->setFormat($format);
+        $this->setContext($context);
     }
```

Comment:
I think removing `$format` and `$context` would have been a BC break, that's why I let them here. Maybe we should introduce some deprecation?